crates/fuel-core/src/database.rs

```diff
@@ -126,10 +126,16 @@ pub type ReyalerIterableKeyValueView = IterableKeyValueView<ColumnType<Relayer>>
 pub type GenesisDatabase<Description = OnChain> = Database<Description, GenesisStage>;
 
 impl OnChainIterableKeyValueView {
-    pub fn latest_height(&self) -> StorageResult<BlockHeight> {
+    pub fn maybe_latest_height(&self) -> StorageResult<Option<BlockHeight>> {
         self.iter_all_keys::<FuelBlocks>(Some(IterDirection::Reverse))
             .next()
-            .ok_or(not_found!("BlockHeight"))?
+            .transpose()
+    }
+
+    pub fn latest_height(&self) -> StorageResult<BlockHeight> {
```

Comment:
Can't we let the caller handle this case?

Reply:
That method existed before, I don't want to affect other places=)